src/arm/decoder.c

```diff
@@ -488,3 +488,43 @@ int ARMDisassemble(struct ARMInstructionInfo* info, uint32_t pc, char* buffer, i
 	buffer[blen - 1] = '\0';
 	return total;
 }
+
+uint32_t ARMResolveMemoryAccess(struct ARMInstructionInfo* info, struct ARMRegisterFile* regs, uint32_t pc) {
+	uint32_t address = 0;
+	int32_t offset = 0;
+	if (info->memory.format & ARM_MEMORY_REGISTER_BASE) {
+		if (info->memory.baseReg == ARM_PC && info->memory.format & ARM_MEMORY_IMMEDIATE_OFFSET) {
+			address = pc & 0xFFFFFFFC;
+		} else {
+			address = regs->gprs[info->memory.baseReg];
+		}
+	}
+	if (info->memory.format & ARM_MEMORY_IMMEDIATE_OFFSET) {
+		offset = info->memory.offset.immediate;
+	} else if (info->memory.format & ARM_MEMORY_REGISTER_OFFSET) {
+		offset = regs->gprs[info->memory.offset.reg];
+	}
+	if (info->memory.format & ARM_MEMORY_SHIFTED_OFFSET) {
+		uint8_t shiftSize = (info->operandFormat & ARM_OPERAND_SHIFT_REGISTER_1) ? regs->gprs[info->memory.offset.shifterReg] : info->memory.offset.shifterImm;
```

Comment:
This check is unneeded; the memory operand can never be shifted by a register, only by an immediate.

Reply:
Oh, I suppose that does make sense, thinking about it long enough. Took several "but what about..." moments before I realized that there really isn't a good use case for generically allowing variable pointer shifting, and in the odd chance you might it you can just do it in separate instruction.